### PR TITLE
feat(cardano-services): stake pools apy and margin averages

### PR DESCRIPTION
**Pull request by jvieiro**
# Context

Provide average APY and margin of all active stake pools for latest epoch at StakePoolProvider

### CODE REVIEW

File: packages/core/src/Provider/StakePoolProvider/types/StakePoolProvider.ts [hunk 1]
```diff
@@ -58,6 +58,10 @@ export interface StakePoolSearchResults {
 }
 
 export interface StakePoolStats {
+  averages: {
+    margin: number;
```
Comment (mkazlauskas) [inline packages/core/src/Provider/StakePoolProvider/types/StakePoolProvider.ts:62]: Please document what the new values are, e.g.

```suggestion
    /** 
     * Within range [0; 1]
     */
    margin: Percent;
```

File: packages/cardano-services/test/StakePool/DbSyncStakePoolProvider/__snapshots__/StakePoolBuilder.test.ts.snap [hunk 2]
```diff
@@ -1518,6 +1518,10 @@ Array [
 
 exports[`StakePoolBuilder queryPoolStats returns active, retired and retiring pools count 1`] = `
 Object {
+  "averages": Object {
+    "apy": 0.0000158041676290942,
+    "margin": 0.000201057770545159,
```
Comment (mkazlauskas) [inline packages/cardano-services/test/StakePool/DbSyncStakePoolProvider/__snapshots__/StakePoolBuilder.test.ts.snap:1523]: This value seems too low, but could be due to the data snapshot. We need to verify that it is a realistic number. cc @iadmytro

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: packages/cardano-services/test/StakePool/DbSyncStakePoolProvider/__snapshots__/StakePoolBuilder.test.ts.snap (hunk 2 above)
Comment (jvieiro) [reply packages/cardano-services/test/StakePool/DbSyncStakePoolProvider/__snapshots__/StakePoolBuilder.test.ts.snap:1523]: Yep, this is due to the db snapshot, there are many missing rows of `pool_update` table. So, a lot of pools have margin = 0

File: packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/queries.ts [hunk 3]
```diff
@@ -821,6 +821,114 @@ FROM last_pool_update AS pool_update
 LEFT JOIN last_pool_retire AS pool_retire 
 	ON pool_update.hash_id = pool_retire.hash_id`;
 
+const findActivePoolsAverages = `
```
Comment (iadmytro) [inline packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/queries.ts:824]: we definitely need tests for this query

**Review by iadmytro — CHANGES_REQUESTED** [review]

**Pull request closed without merging**